Libraries/LLM/LLMModel.swift

```diff
@@ -3,8 +3,48 @@
 import Foundation
 import MLX
 import MLXNN
+import Tokenizers
 
-// Interface for all LLM Models
+/// Container for models that guarantees single threaded access.
+///
+/// Wrap models used by e.g. the UI in a ModelContainer.  Callers can access
+/// the model and/or tokenizer:
+///
+/// ```swift
+/// let promptTokens = await modelContainer.perform { _, tokenizer in
+///     tokenizer.encode(text: prompt)
+/// }
+/// ```
+///
+/// or:
+///
+/// ```swift
+/// let result = await modelContainer.perform { model, tokenizer in
+///     LLM.generate(
+///         promptTokens: promptTokens, parameters: generateParameters, model: model,
+///         tokenizer: tokenizer, extraEOSTokens: modelConfiguration.extraEOSTokens
+///     ) { tokens in
+///     ...
+///     }
+/// }
+/// ```
+public actor ModelContainer {
```

Comment:
This is the most important change -- something that we can use to guard the non-sendable data like models.